### PR TITLE
MachineAuthAPIKey -> SystemUserAPIKey

### PR DESCRIPTION
**Pull request by kobergj**
## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiAccountsHashDifficulty-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/11382/38/1) failed. Further test are cancelled...

File: extensions/settings/pkg/config/parser/parse.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func Validate(cfg *config.Config) error {
 		return shared.MissingJWTTokenError(cfg.Service.Name)
 	}
 
-	if cfg.Metadata.MachineAuthAPIKey == "" {
+	if cfg.Metadata.SystemUserAPIKey == "" {
 		return shared.MissingMachineAuthApiKeyError(cfg.Service.Name)
```
Comment (butonic) [inline extensions/settings/pkg/config/parser/parse.go:41]: could you introduce a new `shared.MissingSystemAuthAPIKey` error?

Comment (butonic) [inline extensions/settings/pkg/config/parser/parse.go:41]: could you introduce a new `shared.MissingSystemAuthAPIKey` error?

File: extensions/sharing/pkg/revaconfig/config.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func SharingConfigFromStruct(cfg *config.Config) map[string]interface{} {
 							"provider_addr":       cfg.UserSharingDrivers.CS3.ProviderAddr,
 							"service_user_id":     cfg.UserSharingDrivers.CS3.ServiceUserID,
 							"service_user_idp":    cfg.UserSharingDrivers.CS3.ServiceUserIDP,
-							"machine_auth_apikey": cfg.UserSharingDrivers.CS3.MachineAuthAPIKey,
+							"system_user_api_key": cfg.UserSharingDrivers.CS3.SystemUserAPIKey,
```
Comment (butonic) [inline extensions/sharing/pkg/revaconfig/config.go:55]:
```suggestion
							"machine_auth_apikey": cfg.UserSharingDrivers.CS3.SystemUserAPIKey,
```
`machine_auth_apikey` is expected by the reva cs3 driver

File: extensions/sharing/pkg/revaconfig/config.go [hunk 3]
```diff
@@ -77,7 +77,7 @@ func SharingConfigFromStruct(cfg *config.Config) map[string]interface{} {
 							"provider_addr":       cfg.PublicSharingDrivers.CS3.ProviderAddr,
 							"service_user_id":     cfg.PublicSharingDrivers.CS3.ServiceUserID,
 							"service_user_idp":    cfg.PublicSharingDrivers.CS3.ServiceUserIDP,
-							"machine_auth_apikey": cfg.PublicSharingDrivers.CS3.MachineAuthAPIKey,
+							"system_user_api_key": cfg.PublicSharingDrivers.CS3.SystemUserAPIKey,
```
Comment (butonic) [inline extensions/sharing/pkg/revaconfig/config.go:80]:
```suggestion
							"machine_auth_apikey": cfg.PublicSharingDrivers.CS3.SystemUserAPIKey,
```
`machine_auth_apikey` is expected by the reva cs3 driver

File: extensions/storage-system/pkg/config/config.go [hunk 4]
```diff
@@ -20,6 +20,7 @@ type Config struct {
 	Reva              *Reva         `yaml:"reva"`
 	MachineAuthAPIKey string        `yaml:"machine_auth_api_key" env:"STORAGE_SYSTEM_MACHINE_AUTH_API_KEY"`
```
Comment (butonic) [inline extensions/storage-system/pkg/config/config.go:21]:
```suggestion
```
no longer needed

File: extensions/storage-system/pkg/config/parser/parse.go [hunk 5]
```diff
@@ -38,7 +38,7 @@ func Validate(cfg *config.Config) error {
 		return shared.MissingJWTTokenError(cfg.Service.Name)
 	}
 
-	if cfg.MachineAuthAPIKey == "" {
+	if cfg.SystemUserAPIKey == "" {
 		return shared.MissingMachineAuthApiKeyError(cfg.Service.Name)
```
Comment (butonic) [inline extensions/storage-system/pkg/config/parser/parse.go:42]: again, use new `MissingSystemAuthAPIKey` error

File: ocis/pkg/init/init.go [hunk 6]
```diff
@@ -191,6 +192,10 @@ func CreateConfig(insecure, forceOverwrite bool, configPath, adminPassword strin
 	if err != nil {
 		return fmt.Errorf("could not generate random password for machineauthsecret: %s", err)
 	}
+	systemUserApiKey, err := generators.GenerateRandomPassword(passwordLength)
+	if err != nil {
+		return fmt.Errorf("could not generate random password for systemuser: %s", err)
```
Comment (butonic) [inline ocis/pkg/init/init.go:197]:
```suggestion
		return fmt.Errorf("could not generate random system user API key: %s", err)
```

**Review by butonic — CHANGES_REQUESTED** [review]

**Review by butonic — APPROVED** [review]

**Pull request merged**